config/routes.js

```diff
@@ -1,24 +1,69 @@
+/***************************************************************************************************
+ ******************************************* dependencies ******************************************
+ **************************************************************************************************/
 const axios = require('axios');
+const express = require('express');
+const db = require('../database/helpers/dbHelpers.js');
+const bcrypt = require('bcryptjs');
+const router = express.Router();
 
-const { authenticate } = require('../auth/authenticate');
+/***************************************************************************************************
+ ******************************************** middleware *******************************************
+ **************************************************************************************************/
+const authenticate = require('../auth/authenticate.js');
+const generateToken = require('../auth/generateToken.js');
 
-module.exports = server => {
-  server.post('/api/register', register);
-  server.post('/api/login', login);
-  server.get('/api/jokes', authenticate, getJokes);
-};
-
-function register(req, res) {
+/***************************************************************************************************
+ ********************************************* Endpoints *******************************************
+ **************************************************************************************************/
+router.post('/register', (req, res) => {
   // implement user registration
-}
+  const newUserCreds = req.body;
+
+  newUserCreds.password = bcrypt.hashSync(
+    newUserCreds.password,
+    db.settings.pwdHashLength
+  );
+
+  db.addUser(newUserCreds)
+    .then(Ids => {
+      db.findByUsername(newUserCreds.username)
+        .first()
+        .then(user => {
+          if (user) {
+            const token = generateToken(user); // Token should result as guest permissions until user verifies (via email)
```

Comment:
You have some of the cleanest routes because of all the abstraction you do, great job.